addons/knative/0.14.x/knative-2.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: ClusterAddon
+metadata:
+  annotations:
+    appversion.kubeaddons.mesosphere.io/knative: 0.14.0
+    catalog.kubeaddons.mesosphere.io/addon-revision: 0.14.0-2
+    values.chart.helm.kubeaddons.mesosphere.io/knative: "https://raw.githubusercontent.com/mesosphere/charts/master/staging/knative/values.yaml"
+  labels:
+    kubeaddons.mesosphere.io/name: knative
+  name: knative
+  namespace: kubeaddons
+spec:
+  kubernetes:
+    minSupportedVersion: v1.15.6
+  cloudProvider:
+    - name: aws
+      enabled: false
+    - name: azure
+      enabled: false
+    - name: docker
+      enabled: false
+    - name: none
+      enabled: false
+  chartReference:
+    chart: knative
+    repo: https://mesosphere.github.io/charts/staging
+    version: 0.3.2
+    values: |
+      eventing:
+        enabled: false
+      eventing-sources:
+        enabled: false
+      serving:
+        enabled: true
+        customMetricsApiservice:
+          enabled: false
+        namespaceKnativeServing:
+          additionalLabels:
+            ca.istio.io/override: "true"
+        configDeployment:
+          registriesSkippingTagResolving: ""
```

Comment:
I think we should add default values here: gcr.io,docker.io and any predictor consumable registries

Reply:
I'd avoid doing that because it can lead to unexpected behavior. I'd suggest keeping addon configuration as close to the default as possible. The default one doesn't set these defaults so shouldn't we (same story as with cluster-local gateway). We can provide cluster.yaml with these defaults set.

Also, if we set _some_ registries here, and a user sets some other registry in cluster.yaml, the default ones will be overwritten and some registries will not be excluded from the resolution which will cause confusion.

Reply:
The knative digest resolution feature is mainly for revision management. It has its value in reality, but fundamentally it contradicts with the way konvoy achieves airgap which leverages the containerd registry mirror support. We should just disable this feature by default. If not configurable, at least we should add `gcr.io,docker.io,mcr.microsoft.com,nvcr.io` to `registriesSkippingTagResolving` as default.

People can hardly hit unexpected behavior unless the predictor image tags got overwritten (less likely in open source community than personal knative user).

This `registriesSkippingTagResolving` config should not be exposed to customers. It is an internal opinion for kfserving service. Even with example config, it would be easily neglected or hard to educate users.

`domain: svc.cluster.local` was incorrect previously. K8s would always return errors because it duplicates the k8s internal dns which is always generated by k8s for every service. `domain` is more like a user facing config.

Reply:
updated